oak-core/src/main/java/org/apache/jackrabbit/oak/query/xpath/Statement.java

```diff
@@ -382,6 +382,18 @@ private static void appendQueryOptions(StringBuilder buff, QueryOptions queryOpt
                             SQL2Parser::escapeStringLiteral));
             optionValues.add("prefetch (" + list + ")");
         }
+        if (queryOptions.hasInsecureOptions()) {
+            List<String> insecureOpts = new ArrayList<>();
+            insecureOpts.add("insecure");
+            if (queryOptions.insecureResultSize) {
+                insecureOpts.add("resultsize");
```

Comment:
I don't like "resultsize" as it's two words concatenated... Options are:
* "resultsize": can be hard to read
* "result_size": looks a bit ugly
* "resultSize": the parser is not case sensitive so it might work
* "result size": this approach is used for "index tag" already

I would prefer "result size" (two words).